packages/store/tests/property-selectors.spec.ts

```diff
@@ -0,0 +1,98 @@
+import { createPropertySelectors } from '../src/selectors/property-selectors';
+
+describe.only('Pick Selector', () => {
+  interface MyStateModel {
+    property1: string;
+    property2: number[];
+  }
+
+  @State<MyStateModel>({
+    name: 'myState',
+    defaults: {
+      property1: 'testValue',
+      property2: [1, 2, 3]
+    }
+  })
+  @Injectable()
+  class MyState {}
+
+  it('Passing null', () => {
+    const slices = createPropertySelectors(null);
+
+    expect(slices).toBe(null);
+  });
+
+  it('Passing undefined', () => {
+    const slices = createPropertySelectors(undefined);
+
+    expect(slices).toBe(null);
+  });
+
+  it('Passing empty object', () => {
+    const slices = createPropertySelectors({});
+
+    expect(slices).toBe(null);
+  });
+
+  it('Should create a selector for each property of state', () => {
+    TestBed.configureTestingModule({
+      imports: [NgxsModule.forRoot([MyState])]
+    });
+
+    const store: Store = TestBed.inject(Store);
+    const myState = store.selectSnapshot<MyStateModel>(MyState);
+    const slices = createPropertySelectors<MyStateModel>(MyState);
+
+    console.log({ slices });
+
+    expect(slices).toHaveProperty('property1');
+    expect(slices).toHaveProperty('property2');
+    // expect(slices).not.toHaveProperty('property3');
+  });
+
+  it('The create selectors should return value of the state', () => {
+    TestBed.configureTestingModule({
+      imports: [NgxsModule.forRoot([MyState])]
+    });
+
+    const store: Store = TestBed.inject(Store);
+    const myState = store.selectSnapshot<MyStateModel>(MyState);
+    const slices = createPropertySelectors<MyStateModel>(MyState);
+
+    expect(slices.property1(myState)).toBe('testValue');
+    expect(slices.property2(myState)).toStrictEqual([1, 2, 3]);
+  });
+
+  it('Should memoise each internal selector', () => {
+    TestBed.configureTestingModule({
+      imports: [NgxsModule.forRoot([MyState])]
+    });
+
+    const store: Store = TestBed.inject(Store);
+    const myState = store.selectSnapshot<MyStateModel>(MyState);
+    const slices1 = createPropertySelectors<MyStateModel>(MyState);
+    const slices2 = createPropertySelectors<MyStateModel>(MyState);
+
+    expect(slices1.property1(myState) === slices2.property1(myState)).toBeTruthy();
+    expect(slices1.property2(myState) === slices2.property2(myState)).toBeTruthy();
+  });
+
+  it('Should memoise each internal selector', () => {
+    TestBed.configureTestingModule({
+      imports: [NgxsModule.forRoot([MyState])]
+    });
+
+    const store: Store = TestBed.inject(Store);
+    const myState = store.selectSnapshot<MyStateModel>(MyState);
+    const slices1 = createPropertySelectors<MyStateModel>(MyState);
+    const slices2 = createPropertySelectors<MyStateModel>(MyState);
+
+    expect(slices1.property1(myState) === slices2.property1(myState)).toBeTruthy();
+    expect(slices1.property2(myState) === slices2.property2(myState)).toBeTruthy();
+  });
```

Comment:
So, I guess the memoisation tests should be these cases:
- Memoisation of the selectors: If I use createPropertySelectors to create a 'slices' object then if I repeatedly ask for the selector for property 1, it should return the same (`===`) selector every time. If createPropertySelectors is called again to create another 'slices' object, then I don't think that the slices object nor the selectors for the same property from each slices object should be equal.
- Memoisation of the values (not fetched again): This is a contrived scenario, but focuses on the memoisation of the property values by the input object. Let's say that you have a prop `foo` that returns a new value every time it is called (we could implement this with a property getter returning a new object every time... or a randomly generated number). Then if the `foo` property selector (which was created by the slicer) is called with the same object twice then it should return the same result (ie. the getter would only ever have been called once).